src/common/WebLink.js

```diff
@@ -1,48 +1,56 @@
 /* @flow strict-local */
 
-import React, { PureComponent } from 'react';
+import React from 'react';
 
-import type { Dispatch } from '../types';
-import { connect } from '../react-redux';
+import invariant from 'invariant';
+import { useSelector } from '../react-redux';
 import Label from './Label';
 import openLink from '../utils/openLink';
-import { getCurrentRealm } from '../selectors';
+import { tryGetCurrentRealm } from '../selectors';
 import { BRAND_COLOR, createStyleSheet } from '../styles';
 
 type Props = $ReadOnly<{|
-  dispatch: Dispatch,
   label: string,
   href: string,
-  realm: URL,
+
+  // What to use to absolutize relative URLs, if different from the
+  // current realm or we don't expect there to be a current realm.
+  baseUrl?: URL,
 |}>;
 
+const componentStyles = createStyleSheet({
+  link: {
+    marginTop: 10,
+    fontSize: 15,
+    color: BRAND_COLOR,
+    textAlign: 'right',
+  },
+});
+
 /**
  * A button styled like a web link.
- *
- * @prop label - Text of the button.
- * @prop href - URL address to open on press.
- * @prop realm - Current realm. Used if the `href` property is relative.
```

Comment:
Since after https://github.com/zulip/zulip-mobile/pull/4606#discussion_r607423727 we're requiring a full, validated URL for the prop (and Flow enforces it), I haven't left such a comment in; let me know if I should do something different here.